src/main/java/frc/robot/sensors/Pigeon.java

```diff
@@ -0,0 +1,27 @@
+package frc.robot.sensors;
+import com.ctre.phoenix.sensors.*;
+import frc.robot.subsystems.DriveBase;
+
+public class Pigeon {
+
+    static PigeonIMU _pidgey;      // Pigeon IMU used to enforce straight drive
+
+    static {
+        //get talon
+        _pidgey = new PigeonIMU(DriveBase.getInstance().leftFrontTalon);
```

Comment:
Have you thought about using a Singleton pattern here instead of initializing this statically? I would recommend a static class for independent calculations, e.g. math routines on passed in variables, but when accessing another instance of a shared resource I think a Singleton has less chance for subtle bugs, e.g. calling this static class from another static variable, since it controls access to the variable through a single instance of an object. Not that this is bad/wrong, just food for thought.

Reply:
Agreed. Let's get the architecture right from early on so that we start on the right foot and everyone can emulate good practices. The 'double instantiation of a single object' issue is a bug we have seen many many times over the years.

Again I think there's a case for the Pigeon to be managed in the DriveBase. I don't think I like the pattern of having a class (robot.sensors.Pigeon) that is this strongly linked to the DriveBase and in particular the internal specifics of the DriveBase (the fact that it has a left front Talon, this should really be private to the DriveBase. One day we might have a Falcon or NEO based DriveBase :) )

Please consider my suggestion of moving the management of the Pigeon to within the drivebase subsystem.